src/ts/navigation/navigation.ts

```diff
@@ -0,0 +1,320 @@
+    headerLinksContainer.appendChild(headerLinksClone);
+
+    navDropdown.appendChild(headerLinksContainer);
+  }
+
+  return navDropdown;
+}
+
+function prepareHtml(containerEl: HTMLElement) {
+  const mainNavEl = getMainNavEl(containerEl);
+
+  const toggleWrapper = document.createElement('div');
+  toggleWrapper.classList.add('cads-greedy-nav');
+
+  const dropdownId = 'cads-greedy-nav-dropdown';
+
+  toggleWrapper.insertAdjacentHTML(
+    'beforeend',
+    buildToggle(containerEl, dropdownId),
+  );
+
+  toggleWrapper.appendChild(buildNavDropdown(dropdownId));
+
+  mainNavEl.insertAdjacentElement('afterend', toggleWrapper);
+}
+
+function setToggleVisibility(containerEl: HTMLElement, breaks: number[]) {
+  const toggleEl = getToggleEl(containerEl);
+
+  if (breaks.length < 1) {
+    toggleEl.parentElement?.setAttribute('aria-haspopup', 'false');
+  } else {
+    toggleEl.parentElement?.setAttribute('aria-haspopup', 'true');
+  }
+}
+
+function openDropDown(containerEl: HTMLElement) {
+  const data = extractDataAttributes(containerEl);
+  const toggleEl = getToggleEl(containerEl);
+  const dropdownEl = getDropdownEl(containerEl);
+
+  toggleEl.setAttribute('aria-expanded', 'true');
+  dropdownEl.setAttribute('aria-hidden', 'false');
+
+  toggleEl.firstElementChild!.textContent = data.labelClose;
+  toggleEl.setAttribute('aria-label', data.ariaLabelClose);
+}
+
+function closeDropDown(containerEl: HTMLElement) {
+  const data = extractDataAttributes(containerEl);
+  const toggleEl = getToggleEl(containerEl);
+  const dropdownEl = getDropdownEl(containerEl);
+
+  toggleEl.setAttribute('aria-expanded', 'false');
+  dropdownEl.setAttribute('aria-hidden', 'true');
+
+  toggleEl.firstElementChild!.textContent = data.label;
+  toggleEl.setAttribute('aria-label', data.ariaLabel);
+}
+
+function toDropdown(containerEl: HTMLElement) {
+  const navDropdown = getDropdownEl(containerEl);
+  const mainNav = getMainNavEl(containerEl);
+
+  if (mainNav && mainNav.children.length > 0 && mainNav.lastElementChild) {
+    if (navDropdown.firstChild) {
+      navDropdown.insertBefore(
+        mainNav.lastElementChild,
+        navDropdown.firstChild,
+      );
+    } else {
+      navDropdown.appendChild(mainNav.lastElementChild);
+    }
+  }
+}
+
+function toMenu(containerEl: HTMLElement) {
+  const navDropdown = getDropdownEl(containerEl);
+  const mainNav = getMainNavEl(containerEl);
+
+  if (
+    mainNav &&
+    navDropdown &&
+    navDropdown.children.length > 0 &&
+    navDropdown.firstElementChild
+  ) {
+    mainNav.appendChild(navDropdown.firstElementChild);
+  }
+}
+
+function computeContentWidth(containerEl: HTMLElement) {
+  const styles = window.getComputedStyle(containerEl);
+  const padding =
+    parseFloat(styles.paddingLeft) + parseFloat(styles.paddingRight);
+
+  return containerEl.clientWidth - padding;
+}
+
+function computeChildrenOffsetWidth(containerEl: HTMLElement) {
```

Comment:
This used to have a 10 pixel offset but the rationale in the comment wasn't clear what benefit it gave. We can add it back if others disagree.